include/aws/common/array_list.inl

```diff
@@ -220,15 +220,23 @@ int aws_array_list_get_at_ptr(const struct aws_array_list *AWS_RESTRICT list, vo
 
 AWS_STATIC_IMPL
 int aws_array_list_set_at(struct aws_array_list *AWS_RESTRICT list, const void *val, size_t index) {
-    size_t necessary_size = (index + 1) * list->item_size;
+    size_t necessary_size;
+    if(!aws_mul_size_checked((index + 1), list->item_size, &necessary_size)) {
```

Comment:
See my discussion on the chat room about adding a macro here.

Reply:
So, we should replace the if statement with
```
AWS_RAISE_ERROR_IF(!aws_mul_size_checked((index + 1), list->item_size, &necessary_size), AWS_OP_ERR)
```
right? Should we implement that way in our last overflow report?